src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -1084,6 +1113,64 @@ private String getWebappImageResource(String imageName) {
     }
     */
     
+    /**
+     * 
+     * @param fileId
+     * @param formatTag
+     * @param formatVersion
+     * @param origin
+     * @param isPublic
+     * @param fileInputStream
+     * @param contentDispositionHeader
+     * @param formDataBodyPart
+     * @return 
+     */
+    @Path("datafile/{fileId}/metadata/{formatTag}/{formatVersion}")
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+
+    public Response saveAuxiliaryFileWithVersion(@PathParam("fileId") Long fileId,
+            @PathParam("formatTag") String formatTag,
+            @PathParam("formatVersion") String formatVersion,
+            @FormDataParam("origin") String origin,
+            @FormDataParam("isPublic") boolean isPublic,
+            @FormDataParam("file") InputStream fileInputStream
+          
+    ) {
+        AuthenticatedUser authenticatedUser;
+        try {
+            authenticatedUser = findAuthenticatedUserOrDie();
+        } catch (WrappedResponse ex) {
+            return error(FORBIDDEN, "Authorized users only.");
+        }
+
+        DataFile dataFile = dataFileService.find(fileId);
+        if (dataFile == null) {
+            return error(BAD_REQUEST, "File not found based on id " + fileId + ".");
+        }
+        
+         if (!permissionService.userOn(authenticatedUser, dataFile.getOwner()).has(Permission.EditDataset)) {
+            return error(FORBIDDEN, "User not authorized to edit the dataset.");
+        }
+
+        if (!dataFile.isTabularData()) {
+            return error(BAD_REQUEST, "Not a tabular DataFile (db id=" + fileId + ")");
+        }
+         
+
+        AuxiliaryFile saved = auxiliaryFileService.processAuxiliaryFile(fileInputStream, dataFile, formatTag, formatVersion, origin, isPublic);
+      
+        if (saved!=null) {
+            return created(saved.getId().toString(),JsonPrinter.json(saved));
```

Comment:
This use of `created` with an id seems to be out of step with all other uses. Notice below that we always return the path to the object just created so it can be further manipulated:

<img width="681" alt="Screen Shot 2020-11-24 at 11 50 22 AM" src="https://user-images.githubusercontent.com/21006/100127138-cd331000-2e4c-11eb-96c4-bb7b0d3e89f0.png">

Since this API is experimental, we could just switch it from `created` to `ok` (both "add file" and "add prov" use `ok` and this code itself used `ok` until I started asking about returning an id in a JSON response which lead to `created` being added in c1e03a0) or we could put a proper path in, like `created` expects. From the test in AccessIT, the file can be downloaded from a path like this:

    .get("/api/access/datafile/" + tabFile1Id + "/metadata/dpJSON/v1")

Reply:
@landreev and I talked this out in Slack and decided to go with switching from created to ok in 471dd25 (which is how this pull request started.